tests/unit/lms/resources/lti_launch_test.py

```diff
@@ -303,130 +303,6 @@ def test_h_user_userid(self, pyramid_request):
-        ):
-            LTILaunchResource(pyramid_request).js_config
-
-    def test_js_config_raises_if_theres_no_tool_consumer_instance_guid(
-        self, pyramid_request
-    ):
-        pyramid_request.params.pop("tool_consumer_instance_guid")
-        pyramid_request.params[
-            "oauth_consumer_key"
-        ] = "Hypothesise3f14c1f7e8c89f73cefacdd1d80d0ef"
-
-        with pytest.raises(
-            HTTPBadRequest,
-            match='Required parameter "tool_consumer_instance_guid" missing from LTI params',
-        ):
-            LTILaunchResource(pyramid_request).js_config
-
-    def test_js_config_contains_one_service_config_for_the_client(self, lti_launch):
-        assert len(lti_launch.js_config["hypothesisClient"]["services"]) == 1
-
-    def test_js_config_includes_the_api_url_for_the_client(self, lti_launch):
-        assert (
-            lti_launch.js_config["hypothesisClient"]["services"][0]["apiUrl"]
-            == "https://example.com/api/"
-        )
-
-    def test_js_config_includes_the_authority_for_the_client(self, lti_launch):
-        assert (
-            lti_launch.js_config["hypothesisClient"]["services"][0]["authority"]
-            == "TEST_AUTHORITY"
-        )
-
-    def test_js_config_disables_share_links_in_client(self, lti_launch):
-        assert (
-            lti_launch.js_config["hypothesisClient"]["services"][0]["enableShareLinks"]
-            is False
-        )
-
-    def test_js_config_includes_grant_token_for_client(self, lti_launch):
-        before = int(datetime.datetime.now().timestamp())
-
-        grant_token = lti_launch.js_config["hypothesisClient"]["services"][0][
-            "grantToken"
-        ]
-
-        claims = jwt.decode(
-            grant_token,
-            algorithms=["HS256"],
-            key="TEST_JWT_CLIENT_SECRET",
-            audience="example.com",
-        )
-        after = int(datetime.datetime.now().timestamp())
-        assert claims["iss"] == "TEST_JWT_CLIENT_ID"
-        assert claims["sub"] == "acct:16aa3b3e92cdfa53e5996d138a7013@TEST_AUTHORITY"
-        assert before <= claims["nbf"] <= after
-        assert claims["exp"] > before
-
-    def test_js_config_includes_the_group_for_the_client(
-        self, lti_launch, pyramid_request
-    ):
-        groups = lti_launch.js_config["hypothesisClient"]["services"][0]["groups"]
-
-        assert groups == [
-            "group:d55a3c86dd79d390ec8dc6a8096d0943044ea268@TEST_AUTHORITY"
-        ]
-
-    def test_js_config_hypothesisClient_config_is_empty_if_provisioning_feature_is_disabled(
-        self, lti_launch, ai_getter
-    ):
-        ai_getter.provisioning_enabled.return_value = False
-
-        assert lti_launch.js_config["hypothesisClient"] == {}
-
-    def test_views_can_mutate_js_config_hypothesisClient_config(self, lti_launch):
-        lti_launch.js_config["hypothesisClient"].update({"a_key": "a_value"})
-
-        assert lti_launch.js_config["hypothesisClient"]["a_key"] == "a_value"
-
```

Comment:
Get outa here! :) The remaining tests are still pretty chunky, but you're in with a chance now